File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        mappings={')':'(','}':'{',']':'['}
        for ele in s:
            if ele in mappings:
                top = stack.pop() if stack else "#"
                if top!=mappings[ele]:
                    return False
            else:
                stack.append(ele)
        return not stack
       

File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = []
    let map = {")":"(","]":"[","}":"{"}
    for (let ch of s){
        if(map[ch]){
            let top=stack[stack.length-1]
            if(top===map[ch]){
                stack.pop()
            }
            else{
                return false
            }
        }
        else{
            stack.push(ch)
        }
    }
    return stack.length===0
};